client/tests/integration/offline_peers.rs

```diff
@@ -2,11 +2,12 @@
 
 use iroha_client::client;
 use iroha_data_model::prelude::*;
-use test_network::*;
+use test_network::{generate_offset, *};
 use tokio::runtime::Runtime;
 
 #[test]
 fn genesis_block_is_commited_with_some_offline_peers() {
+    unique_port::set_offset(generate_offset!()).unwrap();
```

Comment:
OK, so instead of having it be done by adding an extra parameter and letting us control which port goes in, which would allow us to re-factor the way in which we generate the unique port with ease, instead everything is done by mutating the compile-time internal state of `unique_port`. 

Why is this a better idea than just adding a parameter to a function?

Reply:
Do you mean `get_unique_free_port`? My changes are more like extension to `unique_port` crate (and I'd rather made a PR with these changes to the `unique_port`'s repo) - adding a parameter to the function would make it backward-incompatible. Also, I kept it because, per the implementation, it has a global state (I think you mentioned it). But adding a parameter to the function would be a lot better, of course.